### PR TITLE
feat(ilp): configurable socket read retry count 

### PR DESCRIPTION
**Pull request by bluestreak01**
This is socket use is unfair in terms of CPU scheduling but it deals with high load volumes more consistently than "fair" method. By default "fair" method is used and there is no change for end-user.

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 0 / 12 (00.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/AggressiveRecvLineTcpConnectionContext.java|0|9|00.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|0|2|00.00%|

File: core/src/test/resources/server.conf [hunk 1]
```diff
@@ -154,7 +154,7 @@ line.tcp.n.updates.per.load.balance=100000
 line.tcp.max.load.ratio=1.5
 line.tcp.maintenance.job.interval=1000
 line.tcp.default.partition.by=MONTH
-line.tcp.io.aggressive.recv=true
+line.tcp.aggressive.read.retry.count=10000
```
Comment (ideoma) [inline core/src/test/resources/server.conf:157]: In reading in PropServerConfiguration.java default is 0, here default is 1000, I believe they should be same for clarity

Comment (bluestreak01) [reply core/src/test/resources/server.conf:157]: This is test conf file, it doesn't contain defaults.  It is there to make sure we pick up values from file correctly. So all values in this file are different to default :)

**Comment by ideoma** [comment]
#### [PR Coverage check]

:disappointed: **fail** : 3 / 12 (25.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/AggressiveRecvLineTcpConnectionContext.java|0|9|00.00%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|2|2|100.00%|

File: core/src/test/resources/server.conf (hunk 1 above)
Comment (ideoma) [reply core/src/test/resources/server.conf:157]: I'd expect them to be the same to document the defaults to the user and defaults are already optimal.

**Review by ideoma — APPROVED** [review]

File: core/src/test/resources/server.conf (hunk 1 above)
Comment (bluestreak01) [reply core/src/test/resources/server.conf:157]: this is a test file `core/src/TEST/resources/server.conf` it saves no other purpose than to check that settings can be successfully read from the file. The values are deliberately different from the defaults for every setting in this file, not just the line in question.

If this file has same values as default, test won't be able to distinguish where value is coming from, default or file.

I take you point that perhaps `server.conf` that we do deploy should have correct deploy, but that's another file and another change

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 10 / 13 (76.92%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/AggressiveRecvLineTcpConnectionContext.java|8|9|88.89%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|1|1|100.00%|

**Review by ideoma — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 11 / 14 (78.57%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/LineTcpMeasurementScheduler.java|0|1|00.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/DefaultLineTcpReceiverConfiguration.java|1|2|50.00%|
|:large_blue_circle:|io/questdb/cutlass/line/tcp/AggressiveRecvLineTcpConnectionContext.java|8|9|88.89%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|

**Pull request merged**